### PR TITLE
arrow: remove patch for CMake

### PR DESCRIPTION
**Pull request by kou**
Specify library name and version:  **arrow/10.0.0**

CMake find modules provided by Apache Arrow C++ 10.0.0 or later can work with Conan recipes. So we don't need to apply a CMake related patch to Apache Arrow C++ 10.0.0 or later and remove CMake find modules provided by Apache Arrow C++ 10.0.0 or later.

If we don't need to maintain a patch, we can add support for newer versions easily.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/arrow/all/conandata.yml [hunk 1]
```diff
@@ -21,14 +21,11 @@ patches:
...
       patch_description: "use size_t instead of ssize_t"
-      patch_type: "backport"
-    - patch_file: "patches/10.0.0-0002-fix-cmake.patch"
-      patch_description: "use cci package"
       patch_type: "conan"
   "8.0.1":
     - patch_file: "patches/8.0.0-0003-mallctl-takes-size_t.patch"
       patch_description: "use size_t instead of ssize_t"
-      patch_type: "backport"
+      patch_type: "conan"
```
Comment (kou) [inline recipes/arrow/all/conandata.yml:28]: These `patch_type` changes are only for fixing lint error: https://github.com/conan-io/conan-center-index/actions/runs/3764042096/jobs/6398083308

Should we create a separate pull request for them?

Comment (uilianries) [reply recipes/arrow/all/conandata.yml:28]: @kou You can add to the same PR, no problem. Thank you for spotting it!

**Comment by kou** [comment]
Could someone help me?
The error message in https://github.com/conan-io/conan-center-index/pull/14903#issuecomment-1363973457 said `xsimd/9.0.1` isn't found but it seems that `xsimd/9.0.1`: https://conan.io/center/xsimd?version=9.0.1

What should we do to fix this error?

```text
-------- Computing dependency graph --------
xsimd/9.0.1: Not found in local cache, looking in remotes...
xsimd/9.0.1: Checking remote: conan-center-v2
xsimd/9.0.1: Checking remote: c3i_PR-v2-14903
Graph root
    virtual
Requirements
    arrow/8.0.0#cb9c94fff00284fcd2fc2c459e4c29dd - Cache
Graph error
    Package 'xsimd/9.0.1' not resolved: Unable to find 'xsimd/9.0.1' in remotes
```

**Comment by prince-chrismc** [comment]
It's a bug with the bot incorrectly adding the failed label, this checks all passed 🙈 

The v2 pipeline is missing almost all the recipes/binaries so thats completely normal

**Comment by kou** [comment]
Thanks!

Then, this is ready to review!

File: recipes/arrow/all/conanfile.py [hunk 2]
```diff
@@ -392,6 +392,7 @@ def generate(self):
         if is_msvc(self):
             tc.variables["ARROW_USE_STATIC_CRT"] = is_msvc_static_runtime(self)
         tc.variables["ARROW_DEPENDENCY_SOURCE"] = "SYSTEM"
+        tc.variables["ARROW_PACKAGE_KIND"] = "conan"
```
Comment (prince-chrismc) [inline recipes/arrow/all/conanfile.py:395]: What does this do? Oooo

Comment (kou) [reply recipes/arrow/all/conanfile.py:395]: `ARROW_PACKAGE_KIND` is a general Apache Arrow C++ build option to add build metadata.

This is used for handling Conan packages specially in this case:
* https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindjemallocAlt.cmake#L26-L42
* https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindBrotliAlt.cmake#L24-L53

They are done by patching Apache Arrow C++ (see the deleted patch: https://github.com/conan-io/conan-center-index/pull/14903/files#diff-021dc424ccea95d24c75533b53af25e42bb6e51b1ef86d95b1b2c51d3d017ed3 ) before this change. But it increasing maintenance cost of this Conan package. We don't need to patch this Conan package with this approach.

Comment (prince-chrismc) [reply recipes/arrow/all/conanfile.py:395]: Ooooo thank you for the amazing answer ❤️ 

Now can we please stick it in as a comment for future contributions/reviwers? 🙏 

```suggestion
        tc.variables["ARROW_PACKAGE_KIND"] = "conan" # See https://github.com/conan-io/conan-center-index/pull/14903/files#r1057938314 for details
```

Comment (kou) [reply recipes/arrow/all/conanfile.py]: Sure.
I've applied your suggestion.

**Review by prince-chrismc — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**